pkg/kallsyms/kallsyms_test.go

```diff
@@ -224,3 +226,258 @@ func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
+	reader := strings.NewReader(input)
+	addr, mod, err := kernelSymbolAddressFromReader(reader, "test_symbol")
+	require.NoError(t, err)
+	require.Equal(t, uint64(0xffffffffc1234567), addr)
+	require.Equal(t, "test_module", mod)
+}
+
+func TestSymbolExistsWithEmptySymbol(t *testing.T) {
+	exists := SymbolExists("")
+	require.False(t, exists)
+}
+
+// Test for parsing various module formats
+func TestKernelSymbolAddressFromReaderModuleFormats(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		symbol   string
+		wantMod  string
+		wantAddr uint64
+	}{
+		{
+			name:     "no_module",
+			input:    "ffffffffc1234567 T test_symbol",
+			symbol:   "test_symbol",
+			wantMod:  "",
+			wantAddr: 0xffffffffc1234567,
+		},
+		{
+			name:     "with_module",
+			input:    "ffffffffc1234567 T test_symbol\t[test_module]",
+			symbol:   "test_symbol",
+			wantMod:  "test_module",
+			wantAddr: 0xffffffffc1234567,
+		},
+		{
+			name:     "module_with_underscore",
+			input:    "ffffffffc1234567 T test_symbol\t[test_module_name]",
+			symbol:   "test_symbol",
+			wantMod:  "test_module_name",
+			wantAddr: 0xffffffffc1234567,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			addr, mod, err := kernelSymbolAddressFromReader(reader, tt.symbol)
+			require.NoError(t, err)
+			require.Equal(t, tt.wantAddr, addr)
+			require.Equal(t, tt.wantMod, mod)
+		})
+	}
+}
+
+// MockVar represents a mock implementation of ebpf.Var for testing
+type MockVar struct {
+	value interface{}
+}
+
+func (v *MockVar) Set(value interface{}) error {
+	v.value = value
+	return nil
+}
+
+// mockSymbolResolver implements symbolResolver interface for testing
+type mockSymbolResolver struct {
+	resolveFunc func(symbol string) (uint64, error)
+}
+
+func (m *mockSymbolResolver) resolve(symbol string) (uint64, error) {
+	return m.resolveFunc(symbol)
+}
+
+func TestSpecUpdateAddressesEmptySymbols(t *testing.T) {
+	spec := &ebpf.CollectionSpec{
+		Maps:      make(map[string]*ebpf.MapSpec),
+		Programs:  make(map[string]*ebpf.ProgramSpec),
+		Variables: make(map[string]*ebpf.VariableSpec),
+	}
+	err := SpecUpdateAddresses(spec, []string{})
+	require.NoError(t, err)
+}
+
+func TestSpecUpdateAddressesResolverErrors(t *testing.T) {
+	symbolsMap = make(map[string]uint64)
+	triedGetAddr = make(map[string]error)
+
+	failResolver := &mockSymbolResolver{
+		resolveFunc: func(symbol string) (uint64, error) {
+			return 0, fmt.Errorf("mock resolver error")
+		},
+	}
+
+	spec := &ebpf.CollectionSpec{
+		Maps:     make(map[string]*ebpf.MapSpec),
+		Programs: make(map[string]*ebpf.ProgramSpec),
+		Variables: map[string]*ebpf.VariableSpec{
+			"symbol1_addr": {},
+		},
+	}
+
+	err := specUpdateAddresses(
+		[]symbolResolver{failResolver},
+		spec,
+		[]string{"symbol1"},
+	)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "mock resolver error")
+
+	err = specUpdateAddresses(
```

Comment:
Please add a description that this is specifically designed to test if the error cache works. Also would be good to somehow differentiate between the cached error and the non cached one